cli.py

```diff
@@ -305,10 +308,16 @@ def process_file(
     big_docs = split_code(code, language=LANGUAGES[file_ext], chunk_size=10000)
     short_docs = split_code(code, language=LANGUAGES[file_ext], chunk_size=3000)
 
+    # check budget before making GPT-4 calls in extract_code_elements and get_unit_tale
+    session_id = str(uuid.uuid4()) # generate an id for this session
```

Comment:
The `session_id` is being overwritten with each file, which means that users will never exceed the budget when running a folder or a repository.

Additionally, unless the budget is set extremely low, it may be challenging to exceed it with a single file :thinking: